manywheel/Dockerfile_s390x

```diff
@@ -1,18 +1,15 @@
-FROM --platform=linux/s390x docker.io/redhat/ubi9 as base
+FROM --platform=linux/s390x docker.io/ubuntu:24.04 as base
```

Comment:
Note:  If it works, we can stick with ubuntu for now.  Otherwise, let's use `AlmaLinux 8` as mentioned in https://github.com/pypa/manylinux per our discussion.